pkg/operators/ebpf/ebpf.go

```diff
@@ -142,6 +144,11 @@ func (o *ebpfOperator) InstantiateImageOperator(
 		return nil, fmt.Errorf("initializing ebpf gadget: %w", err)
 	}
 
+	err = newInstance.evaluateMapParams(paramValues)
+	if err != nil {
+		return nil, fmt.Errorf("evaluating map params: %w", err)
+	}
```

Comment:
It took me a while to go through the code and verify why the mapIter params needs to be treat differently and be evaluated during the instantiation, instead of during start phase as other ebpf operator's params. IIUC, it's because we want to send to the client operators the value of the mapIter params as ds annotation into the GadgetInfo, which is sent at the end of the instantiation phase (by the svc simple operator). Did I understand correctly?

Reply:
IIUC, yes - all manipulation of datasources and fields should happen in the initialization phase (if possible), so that `GetGadgetInfo()` also knows about and can serialize it (`GetGadgetInfo()` does not call Start/Stop).

Since the annotations can have an influence on how metrics are collected (and by that also change params connected to metrics and possibly even create new data sources from that), it needs to be done in the init phase.

Since you mention that this is different treatment, maybe we should re-check the rest as well 😄 I'll certainly take a look.

Reply:
Exactly. At this point, I would manage all the params in the same way: in the init phase.